src/test/libcephfs/vxattr.cc

```diff
@@ -383,3 +383,37 @@ TEST(LibCephFS, GetAndSetDirRandom) {
 
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, FsCrypt) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  char test_xattr_file[NAME_MAX];
+  sprintf(test_xattr_file, "test_fscrypt_%d", getpid());
+  int fd = ceph_open(cmount, test_xattr_file, O_RDWR|O_CREAT, 0666);
+  ASSERT_GT(fd, 0);
+
+  ASSERT_EQ(0, ceph_fsetxattr(cmount, fd, "ceph.fscrypt.auth", "foo", 3, XATTR_CREATE));
```

Comment:
why did we change this to `XATTR_CREATE`?

Reply:
`XATTR_CREATE` and `CEPH_XATTR_CREATE` are exactly the same thing, just ceph defines it again with a `CEPH_` prefix to support the `_REMOVE` flags.

I just switched it to the `XATTR_CREATE` make it to be consistent with the existing code in `vxattr.cc` file and also will comply with the POSIX.

Reply:
The code for both setxattr -x and rmxattr returns `-CEPHFS_ENODATA` now, you're right they both are the same thing but just to keep thing aligned with the code I thought we should be catching `-CEPHFS_ENODATA`.

Reply:
Good catch. Let me fix it. Thanks @dparmar18